plugins/storage/pyparts/ceph_daemon_checks.py

```diff
@@ -97,6 +96,36 @@ def check_osd_msgr_protocol_versions(self):
             msg = ("{} OSDs do not bind to v2 address".format(len(v1_osds)))
             issue_utils.add_issue(issue_types.CephOSDWarning(msg))
 
+    def get_ceph_bluefs_size(self):
+        """
+        Check if the BlueFS metadata size is too large
+        """
+
+        ceph_osd_df_tree = self.cli.ceph_osd_df_tree_json_decoded()
+        if not ceph_osd_df_tree:
+            return
+
+        bad_meta_osds = []
+        try:
+            for device in ceph_osd_df_tree['nodes']:
+                if device['id'] >= 0:
+                    meta_kb = device['kb_used_meta']
+                    # Usually the meta data is expected to be in 0-4G range
+                    # and we check if it's over 10G
+                    if meta_kb > OSD_META_LIMIT_KB:
+                        bad_meta_osds.append(device['name'])
+        except (KeyError, ValueError):
+            return
+
+        if bad_meta_osds:
+            msg = ("{} OSDs have metadata size larger than 10G. This indicates"
+                   " compaction failure/bug. Possibly affected by "
```

Comment:
i prefer to be consistent about having whitespace at the end of the line rather than the start (makes it easier to spot mistakes)